### PR TITLE
added help when calling bin/cli with no args🤓 😎

### PR DESCRIPTION
**Pull request by Aniket965**
Solved issue #29  && #28 
it now shows help menu if no arguments are present in the call for node/cli
- [x] added help menu if no arguments is provided
- [x] added ipfs check with help is-ipfs module

### CODE REVIEW

**Comment by iamtrask** [comment]
@anoff any chance you'd be around for a quick review?

**Comment by anoff** [comment]
Will take a look within 24h

File: bin/cli.js [hunk 1]
```diff
@@ -41,14 +43,18 @@ program
...
-
-    const ipfsUrl = options.ipfsUrl || 'http://localhost:5001'
-    // conver to object format as expected by ipfs-api
-    const ipfsUrlOpts = url.parse(ipfsUrl)
-    ipfsUrlOpts.host = ipfsUrlOpts.hostname // use only hostname w/o port
-    const ipfs = ipfsAPI(ipfsUrlOpts)
-
-    app.checkForModels(mineAddress, contractAddress, web3, ipfs)
+    if (isIPFS.ipfsUrl(options.ipfsUrl)) {
+      const ipfsUrl = options.ipfsUrl || 'http://localhost:5001'
```
Comment (anoff) [inline bin/cli.js]: Fallback to default should happen outside the if statement. Otherwise it still never activate

File: bin/cli.js [hunk 2]
```diff
@@ -41,14 +43,18 @@ program
...
+      const ipfsUrlOpts = url.parse(ipfsUrl)
+      ipfsUrlOpts.host = ipfsUrlOpts.hostname // use only hostname w/o port
+      const ipfs = ipfsAPI(ipfsUrlOpts)
+      app.checkForModels(mineAddress, contractAddress, web3, ipfs)
+    } else {
+      return console.error('Invalid ipfs-url')
+    }
   })
+program
+  .command('help', {isDefault: true})
```
Comment (anoff) [inline bin/cli.js]: Clever!

**Review by anoff — CHANGES_REQUESTED** [review]

**Comment by anoff** [comment]
Also bonus for emojis in the PR title..🕺

File: bin/cli.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env node
+
 const url = require('url')
 const program = require('commander')
 const app = require('../mine.js')
 const pckg = require('../package.json')
 const Web3 = require('web3')
 const ipfsAPI = require('ipfs-api')
+const isIPFS = require('is-ipfs')
```
Comment (kevinahuber) [inline bin/cli.js:9]: preferred name would be `isIpfs`

File: bin/cli.js [hunk 4]
```diff
@@ -41,14 +43,19 @@ program
...
+    const ipfsUrl = options.ipfsUrl
     // conver to object format as expected by ipfs-api
     const ipfsUrlOpts = url.parse(ipfsUrl)
     ipfsUrlOpts.host = ipfsUrlOpts.hostname // use only hostname w/o port
     const ipfs = ipfsAPI(ipfsUrlOpts)
-
     app.checkForModels(mineAddress, contractAddress, web3, ipfs)
   })
+program
+  .command('help', {isDefault: true})
```
Comment (kevinahuber) [inline bin/cli.js:54]: 🎉

File: bin/cli.js (hunk 3 above)
Comment (Aniket965) [reply bin/cli.js:9]: actually it is used in https://www.npmjs.com/package/is-ipfs examples thats why i used the same

**Comment by anoff** [comment]
Actually I just took a look at the examples page you linked. I'm afraid the module won't help us after all. At this piece of code we're just passing in a URL to an IPFS server. The module seems to validate for hashes though. Our default value would be invalid. Could you change the PR to just validate for a valid HTTP URL for now? I recommend https://www.npmjs.com/package/validator

image

**Comment by Aniket965** [comment]
hey ,  @anoff  , i agree with that that's why i desined the code like that if no args are provided it wont check for ipfs-url validation
```javascript
    if (!options.ipfsUrl) {
      options.ipfsUrl = 'http://localhost:5001'
    } else if (!isIPFS.ipfsUrl(options.ipfsUrl)) {
      return console.error('Invalid ipfs-url')
    }
```

**Comment by anoff** [comment]
Ok that's smart. But still the issue remains that there seems to be a difference in `ipfs url` and what we pass the `ipfs API` as a server URL. If you check how this parameter is used in our current release https://github.com/OpenMined/mine.js/blob/master/.docker/docker-compose.yml it would break the build to merge this PR :(

Sorry I didn't catch this earlier when we talked about using this particular module. Haven't looked at it in depth

**Comment by Aniket965** [comment]
this pr messed up a bit let me close this one

**Pull request closed without merging**